redisinsight/ui/src/pages/rdi/connection-form/ConnectionForm.spec.tsx

```diff
@@ -0,0 +1,131 @@
+import React from 'react'
+
+import { RdiInstance } from 'uiSrc/slices/interfaces'
+import { fireEvent, render, screen, waitFor } from 'uiSrc/utils/test-utils'
+import ConnectionForm, { Props } from './ConnectionForm'
+
+const mockedProps: Props = {
+  onAddInstance: jest.fn(),
+  onCancel: jest.fn(),
+  isLoading: false,
+  editInstance: null
+}
+
+const mockedEditInstance: RdiInstance = {
+  id: '1',
+  name: 'name',
+  url: 'url',
+  username: 'username',
+  password: 'password'
+}
+
+describe('ConnectionForm', () => {
+  it('should render', () => {
+    expect(render(<ConnectionForm {...mockedProps} />)).toBeTruthy()
+  })
+
+  it('should disable submit button when form is invalid', async () => {
+    render(<ConnectionForm {...mockedProps} />)
+
+    // click submit to trigger form validation
+    fireEvent.click(screen.getByRole('button', { name: 'Add Instance' }))
+
+    await waitFor(() => expect(screen.getByRole('button', { name: 'Add Instance' })).toBeDisabled(), {
+      timeout: 1000
+    })
+  })
+
+  it('should disable test connection button when form is invalid', async () => {
+    render(<ConnectionForm {...mockedProps} />)
+
+    // click submit to trigger form validation
+    fireEvent.click(screen.getByRole('button', { name: 'Add Instance' }))
+
+    await waitFor(() => expect(screen.getByRole('button', { name: 'Test Connection' })).toBeDisabled(), {
+      timeout: 1000
+    })
+  })
+
+  it('should not disable submit button when form is valid', async () => {
+    render(<ConnectionForm {...mockedProps} />)
+
+    // click submit to trigger form validation
+    fireEvent.click(screen.getByRole('button', { name: 'Add Instance' }))
+
+    fireEvent.change(screen.getByTestId('connection-form-url-input'), { target: { value: 'url' } })
+    fireEvent.change(screen.getByTestId('connection-form-username-input'), { target: { value: 'username' } })
+    fireEvent.change(screen.getByTestId('connection-form-password-input'), { target: { value: 'password' } })
+
+    await waitFor(() => expect(screen.getByRole('button', { name: 'Add Instance' })).not.toBeDisabled(), {
```

Comment:
You can use act from test-utils. You can look an example in redisinsight/ui/src/pages/home/components/manual-connection/manual-connection-form/ManualConnectionFrom.spec.tsx